reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -140,6 +141,45 @@ static Context of(Object key1, Object value1,
 		return new Context5(key1, value1, key2, value2, key3, value3, key4, value4, key5, value5);
 	}
 
+	/**
+	 * Create a {@link Context} out of a {@link Map}. Prefer this method if you're somehow
+	 * incapable of checking keys are all distinct in other {@link #of(Object, Object, Object, Object, Object, Object, Object, Object)}
+	 * implementations.
+	 *
+	 * @implNote this method compacts smaller maps into a relevant fields-based implementation
+	 * when map size is less than 6.
+	 */
+	static Context of(Map<?, ?> map) {
+		int size = map.size();
```

Comment:
validation from the ContextN() constructor will never work.  
but it is still obvious NPE on the `map`. maybe it is ok)